### PR TITLE
🔄 [CORE-569] Add "Sample Usage" section to modules

### PR DESCRIPTION
**Pull request by eak12913**
This PR adds a new section into the README of the reference documentation for Gruntwork's modules. The new section displays sample terraform source code with the module's source, require and optional variables. This example can be quickly copied to get started with using the module. Each variable lists the description as a comment on top of it.

Looks like this:
image

### CODE REVIEW

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md [hunk 1]
```diff
@@ -73,6 +76,138 @@ The Terraform [instance_refresh](https://registry.terraform.io/providers/hashico
...
+
+<ModuleUsage>
+
+```hcl title="main.tf"
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY GRUNTWORK'S ASG-INSTANCE-REFRESH MODULE
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "asg-instance-refresh" {
```
Comment (brikis98) [inline docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:89]: NIT: use snake case for TF module names. E.g., Convert this to `asg_instance_refresh`.

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md [hunk 2]
```diff
@@ -73,6 +76,138 @@ The Terraform [instance_refresh](https://registry.terraform.io/providers/hashico
...
+
+  source = "git::git@github.com:gruntwork-io/terraform-aws-asg.git//modules/asg-instance-refresh?ref=v0.21.1"
+
+  # ---------------------------------------------------------------------------------------------------------------------
+  # REQUIRED VARIABLES
+  # ---------------------------------------------------------------------------------------------------------------------
+
+  # The name of the ASG.
```
Comment (brikis98) [inline docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:97]: NIT: should the comment specify the type? E.g., This is a `string`, whereas `desired_capacity` is a `number`.

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md [hunk 3]
```diff
@@ -73,6 +76,138 @@ The Terraform [instance_refresh](https://registry.terraform.io/providers/hashico
...
+  asg_name = <INPUT REQUIRED>
+
+  # The desired number of EC2 Instances to run in the ASG initially. Note that auto
+  # scaling policies may change this value. If you're using auto scaling policies to
+  # dynamically resize the cluster, you should actually leave this value as null.
+  desired_capacity = <INPUT REQUIRED>
```
Comment (brikis98) [inline docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:103]: NIT: if an input variable has `sensitive = true`, we should probably render it as commented out, and the comment should say "this is a sensitive variable and should prob be set through env vars."

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md [hunk 4]
```diff
@@ -73,6 +76,138 @@ The Terraform [instance_refresh](https://registry.terraform.io/providers/hashico
 *   Auto Scaling Groups support up to one active instance refresh at a time. When this resource is updated, any existing refresh is cancelled.
 *   Depending on health check settings and group size, an instance refresh may take a long time or fail. The Terraform `instance_refresh` resource does not wait for the instance refresh to complete.
 
+## Sample Usage
+
+<ModuleUsage>
+
+```hcl title="main.tf"
```
Comment (brikis98) [inline docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:83]: NIT: I think you are aware of this, but rendering a tab here with "Terraform" and "Terragrunt" options would be great. I'm guessing you have a ticket to track that as follow-up work somewhere?

**Review by brikis98 — COMMENTED** [review]
First, I wanted to say that this is fantastic to see 🎉 🍻 

Second, I left a few minor comments on one generated usage example... I think the actual fixes likely go into `docs-sourcer`, not this repo, but it was easier to see the output here, so I put my comments here.

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md (hunk 1 above)
Comment (eak12913) [reply docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:89]: ✅

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md (hunk 4 above)
Comment (eak12913) [reply docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:83]: Yes, in fact - that's exactly what's there now just the "Terragrunt" section only has a "Coming soon!" message in it. There's already a [JIRA](https://gruntwork.atlassian.net/browse/CORE-570) to track the work to add in a Terragrunt example.

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md (hunk 3 above)
Comment (eak12913) [reply docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:103]: Captured this in: https://gruntwork.atlassian.net/browse/CORE-578

File: docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md (hunk 2 above)
Comment (eak12913) [reply docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:97]: Captured a design story [here](https://gruntwork.atlassian.net/browse/CORE-579) to track this question. When I discussed this feature with @ebeneliason yesterday, he mentioned this as well. We decided to ship what we had and come back to this. Now we officially have a story to track the work.

Comment (eak12913) [reply docs/reference/modules/terraform-aws-asg/asg-instance-refresh/asg-instance-refresh.md:97]: Update: I changed that story from a design story to an implementation story after discussing with Eben and Ore live.

**Review by oredavids — APPROVED** [review]
Reviewed live.

**Pull request merged**